lib/puppet/type/augeas.rb

```diff
@@ -208,4 +208,39 @@ def sync
     end
   end
 
+  conflict(:file) do
+    cons = []
+    if incl = @parameters[:incl]
+      cons << incl.value
+    end
+
+    if context = @parameters[:context]
+      cons << context.value.gsub(%r{^/files}, '')
+    end
+
+    cons.flatten.uniq
+
+    files = []
+
+    catalog.resources.each do |r| 
```

Comment:
Could you use `catalog.resource(:file).each do |r|` to narrow the list to just file types? That might allow us to later optimize creating that list, separate from using it here.